rfcs/quick-start-book.md

```diff
@@ -0,0 +1,325 @@
+
+Each of these requires their own complementary learning paths that branch as soon as they get to the [Learn page](https://bevyengine.org/learn/) to ensure that the first experience that they have with Bevy matches what they need.
+
+### Bevy Quick Start: the example-first path
+
+Users following the example-first path will tend to take the following route:
+
+1. Encounter the Bevy homepage due to social media or word of mouth.
+2. Navigate to the Learn page.
+3. Choose one of the most relevant **quick start games**.
+4. Complete that tutorial.
+5. Dive into making the game they have in mind, accessing the following resources as needed when they encounter road-blocks:
+   1. Official Examples.
+   2. The Bevy book.
+   3. Community tutorials and template games.
+   4. Various community support forums.
+   5. Streams, YouTube channels and blogs.
+
+Each quick start game should:
+
+1. Assume zero existing knowledge of Bevy.
+2. Begin with a initial high-level explanation of what we're trying to build.
+3. Introduce commented code first, then explain what each of the critical terms means as they come up.
+4. Be broken into compilable, playable sections: one per page of the guide.
+5. Gradually refactor the code to add more functionality.
+6. End with a list of suggestions (with appropriate links) on how you could extend the game in interesting ways.
+
+This path should prioritize:
+
+1. Rapid time-to-fun.
+2. Functional, good-enough explanations that are tied to the code in front of them.
+3. Relevance of quick-start game to the genre of game they want to make.
+4. High asset quality.
+5. Ease of extending the quick-start game with their own tweaks.
+6. Explaining how to get unstuck, through documentation, community help and filing issues.
+
+### The Bevy Book: the definition-first path
+
+Users following the definition-first path will tend to take the following route:
+
+1. Encounter the Bevy homepage due to social media or word of mouth.
+2. Navigate to the Learn page.
+3. Select the **Bevy book**.
+4. Read through the book, largely in order.
+5. Once they feel they have a good enough understanding of the engine, they will begin to make their own games, typically by jumping over onto the example-first path.
+6. As they explore, they will also browse:
+   1. The source code.
+   2. [docs.rs](https://docs.rs/bevy/)
+   3. CONTRIBUTING.md, GitHub issues and pull requests.
+   4. Release notes.
+   5. The engine development channels on Discord.
+
+Each chapter of the Bevy Book should:
+
+1. Have a clear topic, and give a high-level overview of the subtopics it is going to cover and how they fit together.
+2. Be broken down into several sections / pages to focus on detailed topics.
+   1. These should have simple, minimal examples explaining how that functionality works.
+3. Conclude with a compilable and playable example game in a **Bringing it together: {GAME NAME}** section that integrates these topics in a cohesive way.
```

Comment:
I have some concerns about  "Bringing it together: {GAME NAME}":

1. Not all sections will feed well into consolidated example games
2. Writing a "complete game tutorial" for each section is a lot of work for authors
3. Reading/following a "complete game tutorial" for each section is a lot of work for readers
4. Building complete games without pulling in "content dependencies from future sections" will be a challenge. Ex: the "A Bevy of Birds" game will likely want to move the birds, but to do that we'll need to pull in `Time`, which hasn't been covered yet
5. Some sections are clearly tailored around building a thing rather than topical organization. The "Graphics and Assets" section covers Transforms, Hierarchy, 2D (sprites), Assets, Cameras, etc. As we build more features in each of those categories, this section will either balloon in size / scope, or we will need to create new sections (which makes it harder to find content based on topic).
6. Quick Start Games already fill a similar niche (although not identical). Quick Start Games aren't allowed to assume prior knowledge. I think adding "advanced" tutorials that call out prerequisite knowledge would help us fill the "bringing it all together" niche without burdening Bevy Book structure.

I'm currently biased toward the following changes:

1. Remove the "bringing it all together" requirement from sections in favor of small / minimal illustrative examples introduced organically for each concept.
2. Add an "advanced tutorials" concept to cover the "bringing it all together" niche. We could even link to these tutorials from relevant Bevy Book sections.
3.  Give each category/engine pillar its own section. Ex: Getting Started, ECS, Assets, 2D, 3D, UI, Input, Time, Plugins, States, Testing, Rendering. In the interest of Bevy Book flow / not burdening users with advanced concepts too early, we could add "advanced" versions of each section near the end (ex:  `2. ECS ... 8. Advanced ECS`)

Reply:
I agree with these changes. I think we can get a much nicer flow if we're not forced to include cohesive examples throughout (or a strictly linear book order).

The advanced examples sound *great*; it would be wonderful to be able to assume existing knowledge and focus in on more complex features and workflows.

I'll do my best to refactor this now.